src/fidesops/service/connectors/saas_query_config.py

```diff
@@ -89,49 +63,81 @@ def generate_requests(
+        """
+
+        path: str = self.assign_placeholders(current_request.path, param_values)
+
+        headers: Dict[str, Any] = {}
+        for header in current_request.headers or []:
+            headers[header.name] = self.assign_placeholders(header.value, param_values)
+
+        query_params: Dict[str, Any] = {}
+        for query_param in current_request.query_params or []:
+            query_params[query_param.name] = self.assign_placeholders(
+                query_param.value, param_values
+            )
+
+        return SaaSRequestParams(
+            method=current_request.method,
+            path=path,
+            headers=headers,
+            query_params=query_params,
+        )
+
     def generate_query(
         self, input_data: Dict[str, List[Any]], policy: Optional[Policy]
     ) -> SaaSRequestParams:
         """
-        This returns the query/path params needed to make an API call.
+        This returns the header, query, and path params needed to make an API call.
         This is the API equivalent of building the components of a database
         query statement (select statement, where clause, limit, offset, etc.)
         """
-        current_request = self.get_request_by_action("read")
 
-        path: str = current_request.path
-        query_params: Dict[str, Any] = {}
-        body: Optional[str] = current_request.body
-
-        # uses the param names to read from the input data
-        for param in current_request.request_params:
-            if param.type == "query":
-                if param.default_value is not None:
-                    query_params[param.name] = param.default_value
-                elif param.references or param.identity:
-                    query_params[param.name] = input_data[param.name][0]
-            elif param.type == "path":
-                path = path.replace(f"<{param.name}>", input_data[param.name][0])
-            elif param.type == "body":
-                body = SaaSQueryConfig._build_request_body(
-                    path,
-                    param.name,
-                    body,
-                    param.default_value,
-                    input_data[param.name][0] if param.references else None,
-                    input_data[param.name][0] if param.identity else None,
+        current_request: SaaSRequest = self.get_request_by_action("read")
+
+        # create the source of param values to populate the various placeholders
+        # in the path, headers, query_params, and body
+        param_values: Dict[str, Any] = {}
+        for request_param in current_request.request_params:
+            if request_param.references or request_param.identity:
+                param_values[request_param.name] = input_data[request_param.name][0]
+            elif request_param.connector_param:
+                param_values[request_param.name] = pydash.get(
+                    self.secrets, request_param.connector_param
```

Comment:
Are connector params still allowed to have a default value specified in the yaml, the pydantic model seems to indicate this -

Reply:
Removed the unused `ConnectorParams` class and the `default_value` property from `ConnectorParam` (singular). Default values don't make sense now since a static value can be defined inline in the path, headers, query params, or body.